libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,626 @@
+from __future__ import annotations
+
+import enum
+import logging
+import uuid
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+)
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.vectorstores.base import VectorStore
+
+
+class DistanceStrategy(str, enum.Enum):
```

Comment:
believe we already define a DistanceStrategy enum somewhere, would prefer to use that (and validate that it's one of the allowed strategy's in init)